### ISSUE TITLE
If the state is saved while in Half-Life's main menu then the subsequent map load will hang

### ISSUE THREAD

**Issue by YaLTeR** [clementgallet/libTAS#337]
I see this in the terminal:
```
[libTAS f:39] Thread 996 (main) Saved base state of size 41161033 in 0.099399 seconds
[libTAS f:39] Thread 996 (main) Saved state 1 of size 1081673 in 0.025300 seconds
[libTAS f:51] Thread 996 (main) ERROR: Area with soft-dirty cleared but no parent page !?
[libTAS f:51] Thread 996 (main) ERROR: addr >= (current_addr - 4096) failed in getPageFlag
```
Frame 51 is where it hangs I think.

**Comment by clementgallet** [comment]
Mhhh, actually, the code runs fine, and the assert itself was eventually sending a signal, causing a softlock because we don't allow signals when the savestate code is running. The assert condition actually does not matter at all, the case is handled fine. But, the present case where this happen needs some investigation:

The logic behind incremental savestate feature is: for each page we are trying to save, if the base savestate (defined as the first made savestate, and used as a model for all savestates) has this page stored, and there was no writing in this page since, then we can safely not store the page content (only metadata). In every other case, we store the page content.

In the present case, it happened when trying to save the process stack. The base savestate has this region for the stack: `0xfffba000-0xffffe000 ([stack]) with size 278528`. The savestate that we are currently doing has this region for the stack: `0xfffa1000-0xffffe000 ([stack]) with size 380928`.

So, the stack did increase (downward as expected) between the two savestates. The assert case that we are concerned happened during one of the first page of the stack: `0xfffa3000-0xfffa3fff`. We first check if the softdirty bit is set. In this case it was cleared, meaning that there was no writing in this page. Then we check this page in the base savestate. Because in the base savestate the first page of the stack is `0xfffba000-0xfffbafff`, the assert condition was false.

But.... ! The kernel doc says that: 

> the kernel always marks new memory regions (and expanded regions) as soft dirty.

So why the page in question had the softdirty bit cleared, if it was allocated after the base savestate? Also, libTAS is supposed to set all allocated memory to 0, for determinism. This should also set the softdirty bit??? So maybe the kernel has a special treatment for the stack that bypasses all of that?

**Comment by clementgallet** [comment]
DMTCP grows the stack to its [maximum](https://github.com/dmtcp/dmtcp/blob/e4a7a686191983f1bbe0a7fdab063903873cc963/src/processinfo.cpp#L177). We should probably do the same as soon as possible (and init that memory to 0). It should fix determinism for games that rely on uninitialized local variables.

**Comment by clementgallet** [comment]
Fixed in 0463524b890b790eae5983715f8b99675c2e143d
